server/server.go

```diff
@@ -43,22 +50,24 @@ func Start(ip string, port int, cert string, key string, debug bool, t *Tracker,
 		torStr = "Tor"
 	}
 
-	fmt.Printf("%sShuffle Listening on TCP %s:%d (pool size: %d)\n", torStr, ip, port, t.poolSize)
+	log.Infof("%sShuffle Listening on TCP %s:%d (pool size: %d)\n", torStr, ip, port, t.poolSize)
 	for {
 		conn, err := listener.Accept()
 		if err != nil {
 			continue
 		}
 
-		ip, _, _ := net.SplitHostPort(conn.RemoteAddr().String())
+		ip := getIP(conn)
 
 		context, err := limit.Get(nil, ip)
 		if err != nil {
+			log.Debugf(logListener+"Unable to get connection limit: %s\n", err)
```

Comment:
Should we interpolate these prefixes or just keep them concatenated?

I like the way this reads, but it seems odd to use `+` in a Debugf string.

Reply:
I am open to either, just curious what you think =)

Reply:
I tried interpolation first but this was just so much more readable that I went with it.

I think of it in terms of this wise section of PEP20.

```
Readability counts.
Special cases aren't special enough to break the rules. (especially for Go)
Although practicality beats purity. (This)
```
I'll go with whatever you choose.

Reply:
I think the plus pattern is more readable. Good as is!